File: core/Candy/Candy.java
package com.dbc.code;

public class Candy {
    public int candy(int[] ratings) {
        int res = 1, point = 0, pre = 1, max = 1;
        for (int i = 1; i < ratings.length; i++) {
            if (ratings[i] >= ratings[i - 1]) {
                pre = ratings[i] == ratings[i - 1] ? 1 : pre + 1;
                res += pre;
                point = i;
                max = pre;
            } else {
                res += i - point + 1 > max ? i - point + 1 : i - point;
                pre = 1;
            }
        }
        return res;
    }
}

File: core/Candy/Candy.py
from typing import List


def candy(ratings: List[int]) -> int:
    res, point, pre, remain = 1, 0, 1, 1
    for i in range(1, len(ratings)):
        if ratings[i] >= ratings[i - 1]:
            pre = pre + 1 if ratings[i] != ratings[i - 1] else 1
            res += pre
            point = i
            remain = pre
        else:
            res += i - point + 1 if i - point + 1 > remain else i - point
            pre = 1

    return res

if __name__ == '__main__':
    print(candy([1,2,2]))

